grafana/resource_organization.go

```diff
@@ -151,15 +165,15 @@ func ExistsOrganization(d *schema.ResourceData, meta interface{}) (bool, error)
 	return true, err
 }
 
-func ImportOrganization(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+func ImportOrganization(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	exists, err := ExistsOrganization(d, meta)
 	if err != nil || !exists {
 		return nil, errors.New(fmt.Sprintf("Error: Unable to import Grafana Organization: %s.", err))
 	}
 	d.Set("admin_user", "admin")
 	d.Set("create_users", "true")
-	err = ReadOrganization(d, meta)
-	if err != nil {
+	diags := ReadOrganization(ctx, d, meta)
+	if diags != nil {
 		return nil, err
```

Comment:
Unless I'm misunderstanding, I think you'll be returning the wrong `err` here

Reply:
As discussed with @nikosmeds, this is probably just a case where we shouldn't call Read in the Import function. The import should only do the work necessary to set the ID and then the Read will be called implicitly when the state is refreshed following the import. With the read call going away, this function won't have to deal with `Diagnostics`